analyzers/rolaguard_bruteforce_analyzer/BruteforcerMain.py

```diff
@@ -42,20 +42,21 @@ def process_packet(packet, policy):
 
             if len(pot_app_keys) > 0:
                 
-                keys_to_test=list()
-                
-                for pot_app_key in pot_app_keys:
-                    keys_to_test.append(bytes(pot_app_key.app_key_hex.rstrip().upper(), encoding='utf-8')) 
-
+                keys_to_test=[bytes(pk.app_key_hex.rstrip().upper(), encoding='utf-8') for pk in pot_app_keys] 
                 keys_to_test = list(dict.fromkeys(keys_to_test))
+
                 correct_app_keys = LorawanWrapper.testAppKeysWithJoinRequest(
                     keys_to_test,
                     packet.data,
                     dontGenerateKeys = True).split()
```

Comment:
May be it's not necessary, but to avoid problems, you could do a rstrip() and upper() of the correct_app_keys

Reply:
Yes, rstrip and upper are called when the list of keys_to_test is defined.

Reply:
Fixed!